### DISCUSSION TITLE
How to customize user input for LLM prompt?

### DISCUSSION THREAD

**Discussion by lifan0127** [microsoft/teams-sdk#1586] [Q&A] [answered]
Is it possible to customize how user input is represented in a LLM prompt? Currently, I see that text and image files are automatically added to prompt. However, I'd like to customize this behavior to reference attachment files, structured data from message extensions etc. Thanks!

https://github.com/microsoft/teams-ai/blob/main/js/packages/teams-ai/src/prompts/UserInputMessage.ts#L55-L81

**Comment by aacebo** [comment, marked as answer]
no we do not currently offer an easy to use method for overriding any of the prompt sections. Technically if you wanted to the best way would be to extend the `PromptManager` class where they are initialized via the `getPrompt`  method. This is an interesting idea so if you feel strongly that this should be a feature I encourage you to either submit a feature request or create a discussion/PR for this.

**Reply by lifan0127** [reply]
Thanks for your response! While I think there is a need to support more complex, potentially multi-modal input, I don't know the `teams-ai` library well enough to propose a concrete feature request yet.

**Comment by heyitsaamir** [comment]
We're clearing discussions and starting fresh. This discussion has not been acted upon. Please reopen the discussion, or start a new one if this question is still important. Thank you for your understanding.
